test/integration/standalone/other/tls_test.go

```diff
@@ -74,4 +81,203 @@ var _ = g.Describe("Query service_cpm_minute", func() {
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+
+		// Start a new server using the temporary certificate files
+		grpcAddr, httpAddr, tempDeferFn := setup.StandaloneWithTLS(tempCertFile, tempKeyFile)
+		defer tempDeferFn()
+
+		// Create a secure HTTP client with the initial certificates
+		httpClient := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &cryptotls.Config{
+					InsecureSkipVerify: false,
+					RootCAs:            x509.NewCertPool(),
+					MinVersion:         cryptotls.VersionTLS12,
+				},
+			},
+		}
+		// Add the cert to the pool
+		certPool := httpClient.Transport.(*http.Transport).TLSClientConfig.RootCAs
+		certPool.AppendCertsFromPEM(originalCert)
+
+		// Verify HTTP server is working with original certificates
+		healthCheckURL := fmt.Sprintf("https://%s/api/healthz", httpAddr)
+		gm.Eventually(func() error {
+			var respErr error
+			resp, respErr := httpClient.Get(healthCheckURL)
+			if respErr != nil {
+				return respErr
+			}
+			defer resp.Body.Close()
+			if resp.StatusCode != http.StatusOK {
+				return fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+			}
+			return nil
+		}, flags.EventuallyTimeout).Should(gm.Succeed())
+
+		// Create a gRPC connection to populate test data
+		creds, err := credentials.NewClientTLSFromFile(tempCertFile, "localhost")
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+		grpcConn, err := grpchelper.Conn(grpcAddr, 10*time.Second, grpclib.WithTransportCredentials(creds))
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+		defer grpcConn.Close()
+
+		// Populate test data
+		ns := timestamp.NowMilli().UnixNano()
+		testBaseTime := time.Unix(0, ns-ns%int64(time.Minute))
+		casesMeasureData.Write(grpcConn, "service_cpm_minute", "sw_metric", "service_cpm_minute_data.json", testBaseTime, interval)
+
+		// Generate a new certificate with a different CommonName
+		certPEM, keyPEM, err := tls.GenerateSelfSignedCert("updated-localhost", []string{"localhost"})
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+
+		// Update the certificate files in the temporary location
+		err = os.WriteFile(tempCertFile, certPEM, 0o600)
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+		err = os.WriteFile(tempKeyFile, keyPEM, 0o600)
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+
+		// Wait for the server to reload the certificates
+		time.Sleep(1 * time.Second)
+
+		// Create a new HTTP client with updated certificates
+		newHTTPClient := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &cryptotls.Config{
+					InsecureSkipVerify: false,
+					RootCAs:            x509.NewCertPool(),
+					MinVersion:         cryptotls.VersionTLS12,
+				},
+			},
+		}
+		// Add the new cert to the pool
+		newCertPool := newHTTPClient.Transport.(*http.Transport).TLSClientConfig.RootCAs
+		newCertPool.AppendCertsFromPEM(certPEM)
+
+		// Verify HTTP server works with the new certificates
+		gm.Eventually(func() error {
+			var respErr2 error
+			resp, respErr2 := newHTTPClient.Get(healthCheckURL)
```

Comment:
Since the health check is a long-lived connection, please use another endpoint https://github.com/apache/skywalking-banyandb/blob/main/api/proto/banyandb/database/v1/rpc.proto#L104 to verify the HTTP server.